File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math

def isPrime(n):
    if n == 1:
        return False
    
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True
        

def solution(n, k):
    # k진법 변환 및 변환 결과를 str로 변환
    if k < 10:
        temp = ""
        while n >= k:
            r = n % k
            temp = str(r) + temp
            n //= k
        temp = str(n) + temp
        n = temp
    else:
        n = str(n)
        
    candidates = n.split("0")
    
    answer = 0
    for c in candidates:
        if c == "1" or c == "":
            continue
            
        if isPrime(int(c)):
            answer += 1
            
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function isPrime(n) {
    if (n === 1) {
        return false
    }
    
    for (let i = 2; i <= Math.floor(Math.sqrt(n)); i++) {
        if (n % i === 0) {
            return false
        }
    }
    return true
}


function solution(n, k) {
    const nToK = n.toString(k)
    const splitZero = nToK.split("0")
    
    let answer = 0
    for (const sz of splitZero) {
        if (sz === '') {
            continue
        }
        
        if (isPrime(parseInt(sz)) === true) {
            answer += 1
        }
    }
    
    return answer
}